### ISSUE TITLE
Check connectivity in tests

### ISSUE THREAD

**Issue by AntonioLangiu** [measurement-kit/measurement-kit#458]
In a unit test that needs a connection we should test if the connection is available, but I think it should be a strict condition and the test should fail if there is not connection. 

We are testing the connectivity [so](https://github.com/measurement-kit/measurement-kit/blob/master/test/http/stream.cpp#L23):

```
if (CheckConnectivity::is_down()) {
    return;
}
```

but I think we should do it with a REQUIRE codition:

```
REQUIRE ( ! CheckConnectivity::is_down());
```

so if there is not connection we don't have a false positive.

**Comment by bassosimone** [comment]
I reckon that we could argue for long time whether it is more correct to succeed or fail a test requiring the network when the network is required. And I also reckon that the correct way of flagging such a test would probably be to flag it as skipped, which I am not sure whether it's possible to do.

What I fail to reckon is why do you think this is a bug, and what do we gain by failing tests that require the network when the network is down. Consider the case in which you are on a plane running tests and we apply the patch you propose. Then one runs `make check` and then there are tons of failures to investigate and one is certainly going to be angry with you when he/she finds that those tests are not actual failures but rather just that there is no network.

That said, I understand your concern. I think the problem we have here is that we have mixed unit and regress tests, where the former do not need the network and the latter do need it. So, we may possibly want to refactor the tests such that we separate network requiring tests from tests that do not require the network at all. And then allow one to `--disable-network-tests` at configure time.

But also with this change applied, I still think that it would be very surprising to have a failure if you run the tests on a plane and you forgot `--disable-network-tests`, when we can handle the same gracefully by skipping the test and printing an error message.

**Comment by bassosimone** [comment]
Another way to deal with the issue is to write out tests that correctly deal with the network being down and that provides correct checks for both cases. In such event, we could test that the library also behaves correctly when the network is down. Maybe we can consider gradually doing that. Reopening issue with question tag, given that there is also this possibility of proceeding.

**Comment by bassosimone** [comment]
Removing the bikeshedding flag because @AntonioLangiu providing a convincing example: what if there is no connectivity for a glitch on travis and a critical test "passes" because there is no connectivity when instead it should have failed.

So now I think the proper way forward is the following (to be done after 0.2.0):

- [ ] divide tests in tests requiring the network and unit tests
- [ ] try to achieve full coverage with unit tests only
- [ ] remove all the check connectivity stuff and just let tests fail
- [ ] add `./configure` flag to disable network tests

**Comment by bassosimone** [comment]
Now that #897 is merged (and in particulae 38dc666), we can close this issue.
